src/i18n/i18n.service.ts

```diff
@@ -27,14 +32,88 @@ export const replace = (subject, variables) => subject.pipe(
 		const keys = Object.keys(variables);
 		for (const key of keys) {
 			const value = variables[key];
-			while (str.includes(`{{${key}}}`)) {
-				str = str.replace(`{{${key}}}`, value);
-			}
+			str = str.replace(new RegExp(`{{\\s*${key}\\s*}}`, "g"), value);
 		}
 		return str;
 	})
 );
 
+/**
+ * Represents an "overridable" translation value.
+ *
+ * Largely an internal usecase. There are situations where we want an `Observable` that
+ * can emit events from a centralized source **OR** an `Observable` that will emit events
+ * from a component local source. The key example being on/off text in a `Toggle` - In some cases
+ * we want the `Toggle` to use `I18n`s global translations, but in others we'd prefer to use a local
+ * override. We don't ever need to return to a non-overridden state, but we do need the ability to
+ * switch _to_ an overridden sate.
+ */
+export class Overridable {
+	/**
+	 * The raw value of the translation. Defaults to the string value, but will return the value passed to `override`
+	 *
+	 * @readonly
+	 */
+	public get value(): string | Observable<string> {
```

Comment:
Why not also have a setter? Instead of override or could call override.

Reply:
Calling override is more explicit, while assigning to `.value` could be misinterpreted. I'm on the fence.

Reply:
It looks very straightforward to do it and we can do both.

Reply:
I'll add both and we can see which one makes more sense as we use it more ... it's all very subjective